src/ServicePulse.Host/app/js/directives/ui.particular.largeGraph.js

```diff
@@ -1,110 +1,148 @@
+            .range([height - margin, margin]);
+
+        var area = d3.area()
+            .x(function (d, i) {
+                return scaleX(i);
+            })
+            .y(function (d, i) { return scaleY(d); })
+            .y1(function (d) { return scaleY(0); })
+            .curve(d3.curveLinear);
+
+        var line = d3.line()
+            .x(function (d, i) {
+                return scaleX(i);
+            })
+            .y(function (d, i) {
+                return scaleY(d);
+            })
+            .curve(d3.curveLinear);
+
+        chart.append('path')
+            .datum(dataSeries.points)
+            .attr('d', area)
+            .attr('fill', color)
+            .attr('opacity', 0.2)
+            .attr('stroke', color);
+
+        chart.append('path')
+            .datum(Array(dataSeries.points.length).fill(dataSeries.average))
+            .attr('d', line)
+            .attr('stroke', color)
+            .attr('stroke-width', 2)            ;
+
+        chart.selectAll('dot')
+            .data(dataSeries.points)
+            .enter().append('circle')
+            .attr('r', 3)
+            .attr('cx', function (d, i) { return scaleX(i); })
+            .attr('cy', function (d) { return scaleY(d); })
+            .append('svg:title')
+            .text(function (d, i) { return dates[i] + ' | ' + d; });
+            
+        chart.append('g')
+            .attr('class', 'y axis')
+            .attr('transform', 'translate(' + (yAxisAllignedLeft ? margin : graphWidth - margin) + ', 0)')
+            .call((yAxisAllignedLeft ? d3.axisLeft(scaleY) : d3.axisRight(scaleY)));
+
+        var scaleYValue = (scaleY(dataSeries.average == 0 ? 1 : dataSeries.average) + 3) | 0;
+
+        chart.append('text')
+            .attr('x', 0)
+            .attr('y', scaleYValue)
+            .attr('text-anchor', 'start')
+            .attr('font-size', 12)
+            .attr('font-family', 'sans-serif')
+            .text('ø');
+
+        chart.append("text")
+            .attr("text-anchor", "middle")
+            .attr("transform", "translate(" + (margin / 2) + "," + (height / 2) + ")rotate(-90)")  
+            .text("Throughput [msgs/s]");
+
+        chart.append("text")
+            .attr("text-anchor", "middle")  // this makes it easy to centre the text as the transform is applied to the anchor
+            .attr("transform", "translate(" + (graphWidth) + "," + (height / 2) + ")rotate(90)")  // centre below axis
+            .text("Queue Length [msgs]");
+    }
+
     angular.module('ui.particular.largeGraph', [])
         .directive('largeGraph',
             function() {
                 return {
                     restrict: 'E',
-                    scope: {
-                        data: '=plotPoints',
-                        avg: '=plotAverage',
+                    scope: {                        
                         dates: '=xaxisPoints',
+                        firstDataSeries: '=firstDataSeries',
```

Comment:
Is it not better to accept an array of data series?

Reply:
for now the graph is designed to display 2 series one of the reasons is that we display up to 2 Y axis (one on the left and one of the right). I don't know how to handle this in a generic way